poisonivy/init.lua

```diff
@@ -69,33 +64,76 @@ minetest.register_node('poisonivy:climbing', {
 	buildable_to = true,
 })
 
-biome_lib.register_active_spawner({
-	spawn_delay = SPAWN_DELAY,
-	spawn_plants = {"poisonivy:seedling"},
-	avoid_radius = 10,
-	spawn_chance = SPAWN_CHANCE/10,
-	spawn_surfaces = {"default:dirt_with_grass"},
-	avoid_nodes = {"group:poisonivy", "group:flower", "group:flora"},
-	seed_diff = poisonivy_seed_diff,
-	light_min = 7,
-	alt_wallnode = "poisonivy:climbing",
-	verticals_list = walls_list
-})
+local function find_adjacent_wall(pos, verticals, randomflag)
+	local verts = dump(verticals)
+
+	if string.find(verts, minetest.get_node({x=pos.x-1, y=pos.y, z=pos.z  }).name) then return 3 end
+	if string.find(verts, minetest.get_node({x=pos.x+1, y=pos.y, z=pos.z  }).name) then return 2 end
+	if string.find(verts, minetest.get_node({x=pos.x  , y=pos.y, z=pos.z-1}).name) then return 5 end
+	if string.find(verts, minetest.get_node({x=pos.x  , y=pos.y, z=pos.z+1}).name) then return 4 end
+
+	return nil
+end
+
+minetest.register_abm({
+	nodenames = {"default:dirt_with_grass"},
+	interval = 1000,
+	chance = 20,
+	label = "[poisoninvy] spawn plants",
+	min_y = -16,
+    max_y = 48,
+	action = function(pos, node)
+		local p_top = {x = pos.x, y = pos.y + 1, z = pos.z}
+		local n_top = minetest.get_node_or_nil(p_top)
+		if not n_top then return end
+		if n_top.name ~= "air" then return end
+
+		local n_light = minetest.get_node_light(p_top)
+		if n_light < 7 then
+			return
+		end
 
-biome_lib.update_plant({
-	grow_delay = SPAWN_DELAY,
-	grow_chance = GROW_CHANCE,
-	grow_plant = "poisonivy:seedling",
-	grow_result = "poisonivy:sproutling",
-	grow_nodes = {"default:dirt_with_grass"}
+		if minetest.find_node_near(p_top, 10 + math.random(-1.5,2), {"group:poisonivy", "group:flower", "group:flora"}) then
```

Comment:
We should change the range or groups here, with too many mapgen mods generates almost no more poisonivy